backend/bin/kafkaConsumer/common/userCourseProgress/checkAndSendThresholdEmail.ts

```diff
@@ -0,0 +1,63 @@
+require("dotenv-safe").config({
+  allowEmptyValues: process.env.NODE_ENV === "production",
+})
+import { User, Course } from "@prisma/client"
+import { CombinedUserCourseProgress } from "./userFunctions"
+import { KafkaContext } from "../kafkaContext"
+import { v4 as uuidv4 } from "uuid"
+
+interface Props {
+  user: User
+  course: Course
+  combinedUserCourseProgress: CombinedUserCourseProgress
+  context: KafkaContext
+}
+
+export const checkAndSendThresholdEmail = async ({
+  user,
+  course,
+  combinedUserCourseProgress,
+  context,
+}: Props) => {
+  const courseEmailThresholdTemplates = await context.prisma.emailTemplate.findMany(
+    {
+      where: {
+        triggered_automatically_by_course_id: course.id,
+        template_type: "threshold",
+      },
+    },
+  )
+
+  const templatesThatFulfillPoints = courseEmailThresholdTemplates
+    .sort((a, b) => (b.points_threshold ?? 0) - (a.points_threshold ?? 0))
```

Comment:
If this sort is for setting email sending order, I would add a comment to clarify that.

Also, I would change this from descending order to ascending order because if the student is going to receive all matching threshold emails, I think it would be more logical if the last message they receive is the email template with the highest threshold.